### ISSUE TITLE
Adding a new genome

### ISSUE THREAD

**Issue by aakashsur** [higlass/higlass#213]
Hi, we are working with leishmania and have done several HiC experiments, and have so far had a blast using HiGlass. Awesome work!

How would I go about adding a new genome into HiGlass? And where does this have an effect, is it only the search bar (since chromosome labels, grid, and annotations are added seperately)?

![image](https://user-images.githubusercontent.com/7614271/35902675-f7398ac8-0b90-11e8-9ebd-ff7e78ca2aea.png)

**Comment by pkerpedjiev** [comment]
Hey, thanks for opening a new thread about this. So the question of what it means to add a new genome is an open one. HiGlass is designed to be flexible to such definitions and that's why you have the option of adding different chromosome sizes and labels separately.

At its simplest form, a new genome simply consists of a set of chromosome sizes. You can add those using the datatype `chromsizes` and the filetype `chromsizes-tsv`. This will let you add the chromosome labels axis:

![image](https://user-images.githubusercontent.com/2143629/35919159-40e2a762-0be2-11e8-9f4e-c9cb4b45976a.png)

If you want to be able to search using "chromosome coordinates", you have to give the chromsizes file that you uploaded a `coordSystem`. That will let you select it in the search bar dropdown.

Finally, you may want to add gene annotations. That part is a little more complex and requires creating a gene annotations file. There's some documentation on how to do that here: https://hms-dbmi.github.io/higlass-docs/gene_annotations.html

I'm not sure if I answered your question. If not I'll be happy to clarify anything that's unclear.

**Comment by aakashsur** [comment]
Awesome, thanks for the response. Just to clarify a few things, will adding a chromsizes file automatically create a track for the horizontal-chromosome-labels, vertical-chromosome-labels, and 2d-chromosome-grid? 

By adding a coordSystem, do you mean just adding a flag on the ingest, and it would automatically create a coordinate system, or is it a seperate file to ingest?

**Comment by pkerpedjiev** [comment]
Yes to the first question. Whenever you choose to add a track to a given area (e.g. top, left, center, bottom, or right), HiGlass queries the server for a list of available tracks that can be displayed in that area. At the same time, every datatype can be associated with multiple different track types. For example, the `matrix` track type can be displayed as a heatmap in the center area or as a `horizontal-heatmap` in the top area. Likewise, the `chromsizes` datatype can be displayed as a `horizontal-chromosome-labels` track type in the top area, `2d-chromosome-grid` in the center area and as `vertical-chromosome-labels` in the left area.

As for your second question, you just need to add this parameter to the ingest command.

```
  --coordSystem genomeName
```

We're still trying to figure out how to do this best but the reasoning so far has been that the genome position search bar can use two sources of information: chromosome sizes for position search and gene annotations for gene search. These two files will likely have different names but we want to select them at once. And they should both refer to the same "coordinate system", where a coordinate system in this case just refers to an assembly.

Thank you for you questions! Eventually this will make it's way into the docs :)

**Comment by rdacemel** [comment]
Intriguingly custom chromosomes stopped working for me after upgrading to 0.4.17 version, and I had to come back to 0.2.63. I haven't fully figured out when exactly this issue happens, sorry for not being more specific. In short, chromosome grids or labels does not appear anymore and the search is also not working. In contrast, custom annotations keep working although it is not possible to search for genes.

**Comment by pkerpedjiev** [comment]
Strange. A lot has changed between 0.2.63 and 0.4.17. Would be happy to look into it further if you can provide some more information.

**Comment by machinegun** [comment]
Hi,

Was this issue resolved? I am getting the same error. I ingested a new genome with `--coordSystem genomeName`. I can see the coordinate system, but when I try to search particular coordinates, the search doesn't work. Any suggestions on how to make it work?
Thanks!

**Comment by lishuai800** [comment]
Hi, I would like to use Higlass to present the 3D genome of Arabidopsis. However, according to the Docs, there are some obstacles. First, I could not find a refGene file (GenePredEx) for arabidopsis, while the most similar file available now is GenePred file converted from GTF. Does it work to generate gene annotation from genPred file following the instruction. or whether is there a better way to solve this problem? Thanks a lot!

**Comment by pkerpedjiev** [comment]
Shouldn't be a problem to use `genePred`. Just start directly from the "Creating a HiGlass Track" section of the documentation: https://docs.higlass.io/data_preparation.html#creating-a-higlass-track

If you encounter any errors, please feel free to open a new issue or pop into our Slack channel and ask there (http://bit.ly/higlass-slack).
